service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -243,4 +257,24 @@ private String buildPriorPrescription(MedicationRequest medicationRequest) {
         }
         return StringUtils.EMPTY;
     }
+
+    private String buildParticipant(MedicationRequest medicationRequest) {
+        var isPractitioner = buildPredicateReferenceIsA(ResourceType.Practitioner);
+        var isPractitionerRole = buildPredicateReferenceIsA(ResourceType.PractitionerRole);
+        var isOrganization = buildPredicateReferenceIsA(ResourceType.Organization);
+        Predicate<Reference> isRelevant = isPractitioner.or(isPractitionerRole).or(isOrganization);
+
+        if (medicationRequest.hasRecorder() && medicationRequest.getRecorder().hasReference()) {
+            final var reference = medicationRequest.getRecorder();
+            if (reference.hasReferenceElement() && isRelevant.test(reference)) {
```

Comment:
Is `reference.hasReferenceElement()` not the same as `medicationRequest.getRecorder().hasReference()`?

Reply:
I'm just guessing, but this may be because HAPI tries to be "helpful" and create empty instances for you if you call getAbc() for abc that doesn't exist.

Reply:
You're right @n-r-b, it's redundant. Will remove